community/contributing/docs_writing_guidelines.rst

```diff
@@ -274,6 +274,29 @@ element of a list with more than two elements.
 How to write methods and classes
 --------------------------------
 
+Dynamic vs Static typing
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Despite the addition of optional static typing, GDScript remains primarily 
+dynamic language. Unless you are specifily discussing static typing, you
+should only use dynamic typing in your code examples and snipets. 
+This will help reduce confusion to new users by promoting consistent look
+to the code examples they are exposed to. 
+
+**Don't** write using using static typing:
+
+::
+
+    var a := 5
+    var b: String = "My String"
+
+**Do** write code using dynamic typing:
+
+::
+
+    var a = 5
+    var b = "My String"
```

Comment:
Perhaps also share some code samples that seem like they are "static typing" but which actually aren't. For example:

Script classes...

    extends Node
    class_name MyNode # not actually static typing yet. Just makes a global variable

`is` checks...

    if node is MyNode: # compares an object to a Script at runtime
         pass

preloaded constants without a colon...

    const MyNode = preload("res://my_node.gd")

Reply:
I am not sure if class_name is good example here to use it's under static typing in docs so I would assume it's part of static typing way?

Reply:
Hmmm. I suppose it sort of fits the bill for "static typing". It's just a global variable, but because it is evaluated at parse-time, it ends up having requirements that must be met in the "static" editor context rather than runtime.

Reply:
I am not certain adding this will work well it may be to confusing for a new person so for now I will leave it out unless you feel this is absolutely necessary addition?